src/main/scala/tilelink/Broadcast.scala

```diff
@@ -74,72 +173,143 @@ class TLBroadcast(params: TLBroadcastParams)(implicit p: Parameters) extends Laz
     }
   )
 
+  /** Interrupt Node for filter. Each Filter has a interrupt. */
   val intNode = if (params.control.isEmpty) None else Some(IntSourceNode(IntSourcePortSimple(num = 1)))
+  /** Control Node for filter.
+    * [[ProbeFilter.useRegFields]] can be defined to add register to control.
+    */
   val controlNode = params.control.map(x => TLRegisterNode(
     address   = Seq(x.address),
     beatBytes = x.beatBytes,
     device    = new SimpleDevice("cache-controller", Seq("sifive,broadcast0"))))
 
   lazy val module = new LazyModuleImp(this) {
+    /* Number of inward node will decide the number bank. */
     val (ints, fields) = node.in.zip(node.out).zipWithIndex.map { case (((in, edgeIn), (out, edgeOut)), bankIndex) =>
       val clients = edgeIn.client.clients
       val managers = edgeOut.manager.managers
       val lineShift = log2Ceil(params.lineBytes)
 
       import TLBroadcastConstants._
 
+      /** Since [[TLBroadcast]] doesn't implements PutPartialData,
+        * cacheline size must not be less than manger beatBytes.
+        */
       require (params.lineBytes >= edgeOut.manager.beatBytes)
-      // For the probe walker, we need to identify all the caches
-      val caches = clients.filter(_.supports.probe).map(_.sourceId)
-      val cache_targets = caches.map(c => c.start.U)
-
-      // Create the probe filter
-      val flatAddresses = AddressSet.unify(edgeOut.manager.managers.flatMap(_.address))
-      val addressMask = AddressDecoder(flatAddresses.map(Seq(_)), flatAddresses.map(_.mask).reduce(_|_))
+      /** This identify all the caches for the probe walker.  */
+      val caches: Seq[IdRange] = clients.filter(_.supports.probe).map(_.sourceId)
+      /** Use start bit in IdRange to locate which client is target. */
```

Comment:
I think it is the lowest id value in the IdRange, which isn't the same as a bit index within a UInt